src/ds3.c

```diff
@@ -1166,13 +1225,83 @@ static ds3_object _parse_object(xmlDocPtr doc, xmlNodePtr contents_node) {
     return object;
 }
 
+static ds3_search_object* _parse_search_object(xmlDocPtr doc, xmlNodePtr contents_node) {
+    xmlNodePtr child_node;
+    xmlChar* text;
+    ds3_search_object* object = g_new0(ds3_search_object, 1);
+
+    for(child_node = contents_node->xmlChildrenNode; child_node != NULL; child_node = child_node->next) {
```

Comment:
Well at least, what the formatting should be, it doesn't look like I've been very consistent on this myself :\